jbpm-services/jbpm-executor/src/test/java/org/jbpm/executor/impl/wih/MigrationAsyncWorkItemHandlerTest.java

```diff
@@ -0,0 +1,344 @@
+ * limitations under the License.
+ */
+
+package org.jbpm.executor.impl.wih;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.Persistence;
+
+import org.jbpm.executor.ExecutorServiceFactory;
+import org.jbpm.executor.RequeueAware;
+import org.jbpm.executor.impl.ExecutorServiceImpl;
+import org.jbpm.executor.test.CountDownAsyncJobListener;
+import org.jbpm.runtime.manager.impl.DefaultRegisterableItemsFactory;
+import org.jbpm.runtime.manager.impl.deploy.DeploymentDescriptorManager;
+import org.jbpm.runtime.manager.impl.jpa.EntityManagerFactoryManager;
+import org.jbpm.runtime.manager.impl.migration.MigrationException;
+import org.jbpm.runtime.manager.impl.migration.MigrationManager;
+import org.jbpm.runtime.manager.impl.migration.MigrationReport;
+import org.jbpm.runtime.manager.impl.migration.MigrationSpec;
+import org.jbpm.services.task.identity.JBossUserGroupCallbackImpl;
+import org.jbpm.test.listener.process.NodeLeftCountDownProcessEventListener;
+import org.jbpm.test.util.AbstractExecutorBaseTest;
+import org.jbpm.test.util.ExecutorTestUtil;
+import org.jbpm.test.util.PoolingDataSource;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.kie.api.event.process.ProcessEventListener;
+import org.kie.api.executor.ExecutorService;
+import org.kie.api.executor.RequestInfo;
+import org.kie.api.io.ResourceType;
+import org.kie.api.runtime.KieSession;
+import org.kie.api.runtime.manager.RuntimeEngine;
+import org.kie.api.runtime.manager.RuntimeEnvironment;
+import org.kie.api.runtime.manager.RuntimeEnvironmentBuilder;
+import org.kie.api.runtime.manager.RuntimeManager;
+import org.kie.api.runtime.manager.RuntimeManagerFactory;
+import org.kie.api.runtime.manager.audit.AuditService;
+import org.kie.api.runtime.manager.audit.ProcessInstanceLog;
+import org.kie.api.runtime.process.ProcessInstance;
+import org.kie.api.runtime.process.WorkItemHandler;
+import org.kie.api.runtime.query.QueryContext;
+import org.kie.api.task.UserGroupCallback;
+import org.kie.internal.io.ResourceFactory;
+import org.kie.internal.runtime.manager.RuntimeManagerRegistry;
+import org.kie.internal.runtime.manager.context.EmptyContext;
+
+public class MigrationAsyncWorkItemHandlerTest extends AbstractExecutorBaseTest {
+
+    private PoolingDataSource pds;
+    private UserGroupCallback userGroupCallback;  
+    private RuntimeManager manager;
+    private RuntimeManager manager2;
+    private ExecutorService executorService;
+    private EntityManagerFactory emf = null;
+    
+    private EntityManagerFactory emfErrors = null;
```

Comment:
This name might be confusing since it is not really EMF for errors, but for whole jBPM. I would rename emf to emfExecutor and emfErrors to just emf. Does that make sense?